styx-api-service/src/main/java/com/spotify/styx/api/WorkflowResource.java

```diff
@@ -196,12 +197,18 @@ public WorkflowResource(Storage storage) {
     return Response.forPayload(workflowState);
   }
 
-  private Response<List<WorkflowInstanceExecutionData>> instances(String componentId, String endpointId) {
+  private Response<List<WorkflowInstanceExecutionData>> instances(
+      Request request,
+      String componentId,
+      String endpointId) {
+
     final WorkflowId workflowId = WorkflowId.create(componentId, endpointId);
-    final List<WorkflowInstanceExecutionData> data;
+    final String offset = request.parameter("offset").orElse("");
```

Comment:
This is just semantics and I can't really find anything useful on google right now, but isn't this more of a `cursor` than an `offset`? At least in the SQL sense, an `offset` to me is the number of items to skip while this is the element to start from.